File: Npc/11000524.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11000524: Nick
/// </summary>
public class _11000524 : NpcScript {
    protected override int First() {
        return 20;
    }

    // Select 0:
    // $script:0831180407002243$
    // - What is it?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (20, 0):
                // $script:0831180407002245$
                // - Don't sweat it. Crime is pretty bad around here. That's just how it is.
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (20, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11000524.py
""" 11000524: Nick """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 20

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407002243$
        # - What is it?
        return None # TODO

    def __20(self, pick: int) -> int:
        # $script:0831180407002245$
        # - Don't sweat it. Crime is pretty bad around here. That's just how it is.
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (20, 0):
            return Option.CLOSE
        return Option.NONE
